src/prefect/agent/kubernetes/agent.py

```diff
@@ -127,21 +132,12 @@ def __init__(
         )
 
         from prefect.utilities.kubernetes import get_kubernetes_client
-        from kubernetes import client, config
 
-        try:
-            self.logger.debug("Loading incluster configuration")
-            config.load_incluster_config()
-        except config.config_exception.ConfigException as exc:
-            self.logger.warning(
-                "{} Using out of cluster configuration option.".format(exc)
-            )
-            self.logger.debug("Loading out of cluster configuration")
-            config.load_kube_config()
-
-        self.batch_client = get_kubernetes_client("job")
-        self.core_client = get_kubernetes_client("service")
-        self.k8s_client = client
+        # Explicitly do not attempt to load the client from a secret
+        self.batch_client = get_kubernetes_client("job", kubernetes_api_key_secret=None)
+        self.core_client = get_kubernetes_client(
+            "service", kubernetes_api_key_secret=None
+        )
```

Comment:
`get_kubernetes_client` has a default secret `KUBERNETES_API_KEY` since it used by the task library. Setting this to `None` restores the old behavior .